### PR TITLE
Makes batons slowly lose charge when left on

### PR DESCRIPTION
**Pull request by ghost**
Is there really any reason to turn your baton off except when beating someone to death so you don't lose charge?

This takes a long time to lose charge so it's very passive but it's still a factor.

### CODE REVIEW

**Comment by Boredone** [comment]
As a quick estimate, how long do you think it would take for a round-start stun baton to run out with this?

**Comment by dannno** [comment]
:+1: it has always annoyed me that people do this

**Comment by paprka** [comment]
Maybe 30+ minutes but I don't know processing code very well

**Comment by WJohn** [comment]
I feel like this PR has been suggested countless times in the past but was always closed due to various reasons. Is it finally gonna happen this time?

**Comment by Deantwo** [comment]
:+1: 
My only question is.
Does Welding Tools do this too?

**Comment by paprka** [comment]
they always have

**Comment by Deantwo** [comment]
Go for it then! Yay!

**Comment by Fox-McCloud** [comment]
Definitely nerfs the whole "have stun baton at the ready at all times" strategy, which I'm perfectly fine with, to be honest.

:+1:

**Comment by paprka** [comment]
yeah they're way too big

File: code/game/objects/items/weapons/stunbaton.dm [hunk 1]
```diff
@@ -44,7 +44,25 @@
 		else
 			return 0
 
+/obj/item/weapon/melee/baton/proc/update_process()
+	if(status)
+		SSobj.processing.Add(src)
+	else if(!bcell)
+		SSobj.processing.Remove(src)
+	else
```
Comment (Cheridan) [inline code/game/objects/items/weapons/stunbaton.dm]: uh, this else and the else if above it both end up running the same code. so you shouldn't need the else if?

**Comment by Steelpoint** [comment]
To be honest this seems just like a minor annoyance that is going to get someone killed.

I don't have a problem so long as its stated in the Baton description that leaving it on will drain the power, that there's a changelog to tell people of the change and that the wiki is updated to account for this change.

Mainly because every single wiki ever (barring Bay I guess) tells you to leave your Baton on. Its essentially hard coded into people to leave their Batons on to save time.

File: code/game/objects/items/weapons/stunbaton.dm (hunk 1 above)
Comment (paprka) [inline code/game/objects/items/weapons/stunbaton.dm]: hm yeah sure.

**Comment by Alek2ander** [comment]
Can the power per tick while active please not be hardcoded?

**Comment by Deantwo** [comment]
Want a config option for it or what?
why?

**Comment by paprka** [comment]
nah alek just wants a var

**Comment by Deantwo** [comment]
For ingame admin var editing?

**Comment by paprka** [comment]
yeah

**Comment by Dorsisdwarf** [comment]
Does that mean it could be set to a negative number to give someone a recharging stunbaton?

Hey, there's an idea for a new HOS item.

**Comment by paprka** [comment]
that would be fucking hilarious to be honest

**Comment by paprka** [comment]
Ok there's a var for loss per tick you can set on stun baton children if you need to. I don't think making it negative works for auto-recharging, but it's worth a try I guess.

**Comment by Tenebrosity** [comment]
>To be honest this seems just like a minor annoyance that is going to get someone killed.
empty quote

**Comment by paprka** [comment]
And?

Then they learn to not keep their baton on 24/7, batons are really good.

**Comment by Niknakflak** [comment]
I honestly thought this was already a thing and always turned off my baton for this reason. 
-_(._.)_- This is a great change then.

**Comment by palpatine213** [comment]
I forsee harm batonning doubling with this change. "I ran out of charge and couldn't stun him. I had to beat them to crit, it was the only way!"

Also, do borgs lose charge from having their batons out?

**Comment by paprka** [comment]
They sure do

**Comment by paprka** [comment]
...Now they do, anyway, thanks for the tip

**Comment by phil235** [comment]
>This takes a long time to lose charge so it's very passive but it's still a factor.

>Maybe 30+ minutes but I don't know processing code very well

I tested locally, and there's an issue with the MC being super fast (see the fast alien burst issue), because the baton loses all its charge in less than a minute. And the cell drain seems to be exponential The first couple seconds you seem to lose around 15 every second, then 100, then 1000. I'm checking the cell charge var at regular interval and I see 9985, 9960, 9915, 9850, 9330, 7900, 995(less than one batonhit so it turns off).

**Comment by dannno** [comment]
>I forsee harm batonning doubling with this change. "I ran out of charge and couldn't stun him. I had to beat them to crit, it was the only way!"

fuck that, if they ran out of charge because they left their baton on for half an hour that's on them and they can't just go and beat someone to death

**Comment by paprka** [comment]
@phil235 I removed the unnecessary update_icon()

**Comment by Saltycut** [comment]
This is just plain annoying and unecessary. :-1: x :100:

**Pull request merged**